plugins/modules/omevv_firmware_repository_profile.py

```diff
@@ -0,0 +1,440 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+#
+# Dell OpenManage Ansible Modules
+# Version 9.8.0
+# Copyright (C) 2024 Dell Inc. or its subsidiaries. All Rights Reserved.
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#
+
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: omevv_firmware_repository_profile
+short_description: Create, modify and delete firmware repository profile
+version_added: "9.8.0"
+description:
+  - This module allows to create, modify and delete firmware repository profile.
+extends_documentation_fragment:
+  - dellemc.openmanage.omevv_auth_options
+options:
+  state:
+    description:
+      - C(present) creates a new OMEVV firmware repository profile or modifies an existing profile if the profile with the same name already exists.
+      - C(absent) deletes the OMEVV firmware repository profile.
+      - Either I(profile_name) or I(profile_id) is required when I(state) is C(absent).
+    type: str
+    choices: [present, absent]
+    default: present
+  name:
+    description:
+      - Name of the profile.
+      - This is required for modification operation and when I(state) is C(absent).
+    type: str
+  description:
+    description:
+      - Description of OMEVV firmware repository profile..
+    type: str
+  new_name:
+    description: New profile name when modify operation is performed.
+    type: str
+  protocol_type:
+    description:
+      - C(NFS) represents NFS share path.
+      - C(CIFS) represents NFS share path.
+      - C(HTTP) represents HTTP share path.
+      - C(HTTPS) represents HTTPS share path.
+      - This is required when I(state) is C(present) and when creating a new profile.
+    type: str
+    choices: [NFS, CIFS, HTTP, HTTPS]
+  catalog_path:
+    description:
+      - Absolute path of the catalog.
+      - HTTP, HTTPS, NFS, and CIFS paths are supported.
+      - This parameter is required when I(state) is C(present).
+    type: str
+  share_username:
+    description:
+      - Username of the share.
+      - This is required when I(catalog_path) is HTTPS or CIFS.
+    type: str
+  share_password:
+    description:
+      - Password of the share.
+      - This is required when I(catalog_path) is HTTPS or CIFS.
+    type: str
+  share_domain:
+    description: Domain of the share.
+    type: str
+requirements:
+  - "python >= 3.9.6"
+author:
+  - "Shivam Sharma(@ShivamSh3)"
+attributes:
+    check_mode:
+        description: Runs task to validate without performing action on the target machine.
+        support: full
+    diff_mode:
+        description: Runs the task to report the changes made or to be made.
```

Comment:
Runs the task to report the changes made or the changes that must be applied.

Reply:
done